trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid_state.F

```diff
@@ -0,0 +1,76 @@
+! This module defines and instantiates objects
+! for a hybrid type reservoir's state.
+! State holds and tracks dynamic/changing variables
+! that are only relevant to the given hybrid
+! reservoir object and not other modules or areas
+! of the system.
+
+module module_persistence_levelpool_hybrid_state
+
+    use module_levelpool, only: levelpool_type
+    use module_reservoir_base
+    implicit none
+
+    ! Extend/derive hybrid state from the abstract base
+    ! type for reservoir state.
+    type, extends(reservoir_base_state_type) :: hybrid_state_interface
```

Comment:
how is "hybrid_state_interface" "pure". Pure, in fortran, means it has no side effects. I'm dont think that's what you mean. 
I still dont like the conventions. To me, the indication of an extension would be
`type, extends(reservoir_state_) :: reservoir_state_hybrid`
when would I care about or use the distinction between "type" and "interface", which I dont understand. 
Also, `reservoir_base_state_type`? That makes my head hurt.

Reply:
Types tagged with interface are pure virtual classes in the C++ sense, they should never be instantiated and only used as base for a derived type.  Ending the type name with interface allows real types to immediately be distinguishable from bases that exist only to ensure a type implements a required interface.  Now using interface as tag is not necessary but I consider it beneficial to be able to distinguish normal and pure virtual types.